### ISSUE TITLE
Update Cody extension name for SEO

### ISSUE THREAD

**Issue by chillatom** [sourcegraph/cody#3559]
See [link for context](https://docs.google.com/document/d/1Dd6IegY0l7g2x2xBPFKFuEtKw6U0F6uYGAJKTK6AwUQ/edit#heading=h.7g5r2vmzz2jp)

Update extension marketplace name and description

**Comment by kalanchan** [comment]
@toolmantim to follow up with @chillatom on implications of changing the name in other areas of the product

**Comment by toolmantim** [comment]
I can see that Tabnine do this too for the name:

![Image](https://github.com/sourcegraph/cody/assets/153/5a9fc862-0246-4897-a260-e9b23c1a0fef)

As far as UI footprint implications of this change, I gave it a test on a branch and it does make our VS Code notifications spammy:

![Image](https://github.com/sourcegraph/cody/assets/153/cc45d8c8-1d3f-49da-b178-fa6db9a616ad)

![Image](https://github.com/sourcegraph/cody/assets/153/bf5ea49c-26bd-4637-ad22-b182eca1f8c9)

![Image](https://github.com/sourcegraph/cody/assets/153/338dfc94-421f-4761-ac18-33065cbd25d3)

I personally think this change looks gross, and Tabnine is often quoted by @sqs as an anti-example of what we want to do. e.g. in the status bar we used "Sign In" instead of "Sign in to use Cody", which is a much more subtle change than this.

I can see an argument for trying to temporarily to see if we do get heaps more traffic from the VS Code marketplace. But unless this is going to have a huge impact, and I'd strongly suggest we don't do it.
